### PR TITLE
☎️ Refactor Nexmo

### PR DESCRIPTION
**Pull request by rowanseymour**
* Create a voice application per channel
* Move number management into a client class in the nexmo channel type package
* Delete nexmo applications when releasing a channel

### CODE REVIEW

**Comment by nicpottier** [comment]
Generally we do not delete applications on a client's account, because we don't know if they may have changed the URLs themselves at some point and they are in use. Safer to just leave turds in their account than actually delete something that may be in use.

**Comment by rowanseymour** [comment]
@nicpottier I can remove it but I was copying the other channel types and we do delete created applications for at least Twilio and Plivo

File: temba/channels/handlers.py [hunk 1]
```diff
@@ -34,6 +34,13 @@ class BaseChannelHandler(View):
 
     @csrf_exempt
     def dispatch(self, request, *args, **kwargs):
+        if not settings.TESTING:  # pragma: no cover
```
Comment (rowanseymour) [inline temba/channels/handlers.py]: only channel handlers left in RP are IVR handlers which we shouldn't ever get to now except in tests

**Comment by nicpottier** [comment]
Huh, that they are, well apparently I'm just losing my mind entirely, carry on.

**Comment by rowanseymour** [comment]
@nicpottier one remaining question for this PR (or one after it) is what to do with existing Nexmo channels. It does appear Nexmo IVR has been broken for a while so that might be an argument for not touching existing channels, but we could go thu all Nexmo channels, lookup the number to see if it supports voice, create a new voice application with the correct mailroom URLs.

Or at the very least, go thru the org configs and clean out the keys that no longer exist (Nexmo NEXMO_APP_ID, NEXMO_APP_SECRET, NEXMO_UUID)

**Comment by nicpottier** [comment]
This is affects incoming IVR which isn't used that much obviously or we would have heard from people, so no, don't think we need to do anything, can just have them reconnect if it does come up.

**Comment by rowanseymour** [comment]
@nicpottier clean up the unused keys from org config? Just thinking that even if we changed our mind in the future about fixing those apps, the app details have been copied into channel.config anyways.

**Comment by nicpottier** [comment]
We could, just don't see a whole lot of benefit to it. Up to you.

**Comment by rowanseymour** [comment]
Meh I'll leave it then

**Pull request merged**